backend/app/main.py

```
@@ -0,0 +1,108 @@
+def delete_note(note_id: int, db: Session = Depends(get_db)):
+    db_note = db.query(models.Note).filter(models.Note.id == note_id).first()
+    if db_note is None:
+        raise HTTPException(status_code=404, detail="Note not found")
+    
+    db.delete(db_note)
+    db.commit()
+    return {"message": "Note deleted"}
+
+# Bookmarks endpoints
+@app.post("/bookmarks/", response_model=schemas.Bookmark)
+def create_bookmark(bookmark: schemas.BookmarkCreate, db: Session = Depends(get_db)):
```

Ruff (0.13.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)